src/bacnet/basic/service/s_cov.c

```diff
@@ -106,13 +106,22 @@ uint8_t Send_COV_Subscribe(
     int len = 0;
     int pdu_len = 0;
     int bytes_sent = 0;
+#if BACNET_SEGMENTATION_ENABLED
+    uint8_t segmentation = 0;
+    uint16_t maxsegments = 0;
+#endif
     BACNET_NPDU_DATA npdu_data;
 
     if (!dcc_communication_enabled()) {
         return 0;
     }
     /* is the device bound? */
-    status = address_get_by_device(device_id, &max_apdu, &dest);
+    status = address_get_by_device(
```

Comment:
this change is not needed if `address_segment_get_by_device()` is added to address.c module and address_get_by_device() remains unchanged